### PR TITLE
📖  added packet where needed across the cluster-api

### PR DESCRIPTION
**Pull request by gianarb**
The Packet provider was missed from a small number of pages in the
documentation.

/cc. @deitch

### CODE REVIEW

File: docs/book/src/user/quick-start.md [hunk 1]
```diff
@@ -228,6 +228,11 @@ clusterctl init --infrastructure openstack
 
 Please visit the [Metal3 project][Metal3 getting started guide].
 
+{{#/tab }}
+{{#tab Packet}}
```
Comment (fabriziopandini) [inline docs/book/src/user/quick-start.md:232]: Packet should be added to the list of tabs at line 130

Comment (gianarb) [reply docs/book/src/user/quick-start.md:232]: Can I change it now? Because you approved it, not sure if that will drive the automation crazy or not

**Comment by fabriziopandini** [comment]
@vincepri this one is ready to go for me
over to you

File: docs/book/src/clusterctl/provider-contract.md [hunk 2]
```diff
@@ -174,6 +174,7 @@ providers.
 |CAPD          | cluster.x-k8s.io/provider=infrastructure-docker    |
 |CAPM3         | cluster.x-k8s.io/provider=infrastructure-metal3    |
 |CAPZ          | cluster.x-k8s.io/provider=infrastructure-azure     |
+|CAPP          | cluster.x-k8s.io/provider=infrastructure-packet     |
```
Comment (CecileRobertMichon) [inline docs/book/src/clusterctl/provider-contract.md:177]: this should be in alphabetical order

File: docs/book/src/user/quick-start.md (hunk 1 above)
Comment (CecileRobertMichon) [inline docs/book/src/user/quick-start.md:232]: Does Packet also need to be added to line 491 (CNI instructions) https://github.com/kubernetes-sigs/cluster-api/pull/3123/files#diff-dccb64095f5c5a09d32f119446301d12L486?

File: docs/book/src/user/quick-start.md [hunk 3]
```diff
@@ -483,8 +488,8 @@ kubectl --namespace=default get secret/capi-quickstart-kubeconfig -o jsonpath={.
 
 Calico is used here as an example.
 
-{{#tabs name:"tab-deploy-cni" tabs:"AWS|Docker|GCP|vSphere|OpenStack|Metal3,Azure"}}
-{{#tab AWS|Docker|GCP|vSphere|OpenStack|Metal3}}
+{{#tabs name:"tab-deploy-cni" tabs:"AWS|Docker|GCP|vSphere|OpenStack|Metal3,Azure,Packet"}}
```
Comment (fabriziopandini) [inline docs/book/src/user/quick-start.md:491]: If packet uses calico like all the other providers (except Azure) this should be
```suggestion
{{#tabs name:"tab-deploy-cni" tabs:"AWS|Docker|GCP|vSphere|OpenStack|Metal3|Packet,Azure"}}
```

**Comment by gianarb** [comment]
@CecileRobertMichon can you have another look?
Thanks a lot

**Comment by vincepri** [comment]
@gianarb we usually have 1 commit, if all the changes are part of the same patchset

**Comment by vincepri** [comment]
Changes LGTM, over to you @CecileRobertMichon for a final look

**Pull request merged**